src/main/java/org/cirdles/topsoil/dataset/Dataset.java

```diff
@@ -33,6 +33,8 @@
 
     public Optional<String> getName();
 
+    public void setName(String name);
```

Comment:
I don't like making this mutable. I had plans for making the `DatasetReader`s put the filename as the name.

A better solution than adding this would be to modify `getName` to always return a string and add name to the `SimpleDataset` constructor. Of course, then the readers have to take this into account and might sometimes need to generate a name.